### ISSUE TITLE
Compiler panics when accessing a field of a temporary

### ISSUE THREAD

**Issue by jDomantas** [mun-lang/mun#225]
Trying to compile this:

```
struct Num {
    value: i64,
}

pub fn foo() {
    Num { value: 0 }.value;
}
```

results in

```
thread 'main' panicked at 'internal error: entered unreachable code: invalid place expression', <::std::macros::panic macros>:5:6
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```

**Comment by baszalmstra** [comment]
Thats a good find! I did not think of trying that. I'll take a look asap! :)

**Comment by baszalmstra** [comment]
Took me a while but I finally created a PR that fixes this issue (#232)
